In [ ]:
import pandas as pd

df = pd.read_csv(
    "../data/DSI_kickstarterscrape_dataset.csv",
    encoding="latin1"
)

print("Dataset shape:", df.shape)

Dataset shape: (41965, 17)


In [17]:
print("Columns:")
print(df.columns.tolist())

print("\nFirst 20 location values:")
print(df["location"].head(20))

print("\nMissing location values:")
print(df["location"].isna().sum())

Columns:
['project id', 'name', 'url', 'category', 'subcategory', 'location', 'status', 'goal', 'pledged', 'funded percentage', 'backers', 'funded date', 'levels', 'reward levels', 'updates', 'comments', 'duration']

First 20 location values:
0        Columbia, MO
1       Maplewood, NJ
2     Los Angeles, CA
3            Novi, MI
4        Portland, OR
5     Collegedale, TN
6         Chicago, IL
7         Chicago, IL
8         Chicago, IL
9             Unknown
10        Ashland, OR
11       New York, NY
12            Unknown
13        Chicago, IL
14     Washington, DC
15          Boise, ID
16          Tampa, FL
17         Austin, TX
18       New York, NY
19         Denver, CO
Name: location, dtype: str

Missing location values:
0


In [18]:
print(df["location"].dropna().sample(20, random_state=42).to_string(index=False))

  Rollinsford, NH
      Chicago, IL
  Los Angeles, CA
     Damascus, OR
 Newport News, VA
San Francisco, CA
        Provo, UT
 Waterloo, Canada
    Knoxville, TN
San Francisco, CA
 Philadelphia, PA
  New Orleans, LA
     Brooklyn, NY
  Los Angeles, CA
  Los Angeles, CA
     New York, NY
  Los Angeles, CA
 Laguna Hills, CA
  New Orleans, LA
    Anchorage, AK


In [19]:
us_states = {
    "AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE",
    "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS",
    "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS",
    "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY",
    "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC",
    "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV",
    "WI", "WY", "DC"
}

location_state = (
    df["location"]
    .fillna("")
    .str.rsplit(",", n=1)
    .str[-1]
    .str.strip()
)

us_df = df[location_state.isin(us_states)].copy()

print("Original records:", len(df))
print("U.S. records:", len(us_df))
print("Records not identified as U.S.:", len(df) - len(us_df))

Original records: 41965
U.S. records: 38504
Records not identified as U.S.: 3461


In [20]:
print("U.S. dataset shape:", us_df.shape)

print("\nUnique U.S. location values:")
print(us_df["location"].nunique())

print("\nSample of U.S. locations:")
print(us_df["location"].sample(20, random_state=42).to_string(index=False))

U.S. dataset shape: (38504, 17)

Unique U.S. location values:
3772

Sample of U.S. locations:
     Portland, OR
       Boston, MA
      Chicago, IL
    Nashville, TN
       Dallas, TX
      Seattle, WA
  Idaho Falls, ID
        Tulsa, OK
     New York, NY
      Atlanta, GA
San Francisco, CA
     Sherwood, OR
    Vancouver, WA
  Los Angeles, CA
   Somerville, MA
  Los Angeles, CA
     Sterling, VA
 Philadelphia, PA
      Detroit, MI
       Denver, CO


In [21]:
# Verify that all filtered locations end with a valid U.S. state/DC abbreviation

us_location_states = (
    us_df["location"]
    .str.rsplit(",", n=1)
    .str[-1]
    .str.strip()
)

invalid_states = sorted(set(us_location_states) - us_states)

print("Invalid state values:", invalid_states)
print("Number of invalid state values:", len(invalid_states))

Invalid state values: []
Number of invalid state values: 0


In [22]:
print("Sample of U.S.-based campaigns:")
print(us_df[["name", "location", "status"]].head(20).to_string(index=False))

Sample of U.S.-based campaigns:
                                                            name          location     status
                                           WHILE THE TREES SLEEP      Columbia, MO successful
                            Educational Online Trading Card Game     Maplewood, NJ     failed
     GETTING OVER - One son's search to finally know his father.   Los Angeles, CA successful
The Launch of FlyeGrlRoyalty &quot;The New Name In Fashion&quot;          Novi, MI     failed
     Dinner Party - a short film about friendship... and revenge      Portland, OR successful
                                                           Mezzo   Collegedale, TN     failed
         Help APORTA continue to make handwoven/knit accessories       Chicago, IL successful
                                         Music - Comedy - Album!       Chicago, IL successful
                                         The Apocalypse Calendar       Chicago, IL successful
                            

In [23]:
# Save the U.S.-filtered dataset for Task 5
us_df.to_csv("../data/kickstarter_us_filtered.csv", index=False)

print("Saved U.S.-filtered dataset!")
print("Final dataset shape:", us_df.shape)

Saved U.S.-filtered dataset!
Final dataset shape: (38504, 17)




# Task 4: Prepare U.S.-Based Campaign Records

### Location Review

Task 4 uses the cleaned dataset produced by Task 3, `kickstarter_missing_values_handled.csv`, rather than the original Kickstarter dataset. The `location` column was reviewed to understand how campaign locations are represented. Location values generally follow a `City, State/Country` format, such as `Los Angeles, CA` and `Melbourne, Australia`.

Records without an identifiable U.S. location were excluded from the U.S.-based subset.

### U.S.-Based Campaign Filtering

To identify U.S.-based campaigns, the final component of each location value was extracted and compared against a list containing the 50 U.S. state abbreviations and Washington, D.C.

The filtered dataset contains:

* **U.S.-based campaigns:** 38,504 records
* **U.S. dataset shape:** 38,504 rows × 17 columns

The resulting `us_df` DataFrame contains campaign records whose location ends with a recognized U.S. state or Washington, D.C. abbreviation.

### Location Information Review

The filtered records were reviewed using the campaign name, location, and status fields.

The original `location` field was retained rather than creating duplicate city and state columns. This preserves the location information for potential use in future exploratory analysis.

A sample of the filtered records was reviewed using the campaign name, location, and status fields. The sample included U.S.-based locations such as Columbia, MO; Los Angeles, CA; Chicago, IL; Washington, DC; and Austin, TX. Both successful and failed campaigns were present in the filtered dataset.

### Verification

The U.S.-based dataset was verified by checking the state/DC portion of each retained location against the list of recognized U.S. state and D.C. abbreviations.

**Verification results:**

* **U.S. dataset shape:** 38,504 rows × 17 columns
* **Invalid state values:** 0 (confirm this with your updated verification output)

### Output for Task 5

The U.S.-filtered dataset was saved as `kickstarter_us_filtered.csv` for use in Task 5, where categorical variables will be prepared for machine learning.

The resulting dataset is ready for the next stage of data preparation and future exploratory analysis.
